lms/views/basic_lti_launch.py

```diff
@@ -45,7 +45,7 @@ def sync_lti_data_to_h(self):
         and group corresponding to the LTI user and course.
         """
 
-        self.request.find_service(name="lti_h").single_group_sync()
+        self.request.find_service(name="lti_h").sync([self.context.h_group])
```

Comment:
This makes sense. `context` is a web-layer thing. It's right for views to access `context`. `context` is not a services-layer thing, services shouldn't be accessing it themselves.

Reply:
Apart from anything else, it kills off a fairly content free method.